python_dependency_verifier/python_dependency_verifier.py

```diff
@@ -137,63 +53,150 @@ def get_latest_version_number(package_name):
     return None
 
 
-def check_all_filename_in_subdirs(path, filename_to_search_for,
-                                  field_dependency_name, ignore_this):
-    """receives a path and setup filename to search for,
-    a field dependency name and regex to ignore
-    prints filename and the list of dependencies it uses,
-    whether they are version locked,
-    which version and the latest version of the package available on pip
-    """
-    for dir in os.listdir(path):
-        try:
-            filename = path + dir + filename_to_search_for
-            file_contents = get_file_contents(filename)
-            file_contents = file_contents.replace(" = ", "=")
-            if field_dependency_name not in file_contents:
-                continue
-            dependency_list = get_dependency_with_latest_version(
-                get_install_requires_field_contents(file_contents),
-                ignore_this)
-            dependency_list += get_append_list(file_contents, ignore_this)
-            if dependency_list is not []:
-                print filename
-                print dependency_list
-
-        except IOError:
-            continue
-
-
-def check_all_filename_in_subdirs(path, filename_to_search_for,
-                                  field_dependency_name, ignore_this):
-    """receives a path and setup filename to search for,
-    a field dependency name and regex to ignore
-    returns a dict of filename and the list of dependencies it uses,
-    whether they are version locked,
-    which version and the latest version of the package available on pip
-    """
-    result = []
-    for dir in os.listdir(path):
-        try:
-            filename = path + dir + filename_to_search_for
-            file_contents = get_file_contents(filename)
-            file_contents = file_contents.replace(" = ", "=")
-            if field_dependency_name not in file_contents:
-                continue
-            dependency_list = get_dependency_with_latest_version(
-                get_install_requires_field_contents(file_contents),
-                ignore_this)
-            dependency_list += get_append_list(file_contents, ignore_this)
-            if dependency_list is not []:
-                result.append([filename, dependency_list])
-
-        except IOError:
-            continue
-
-    return result
+def fix_unequal_conditions(string):
+    """makes <> conditions human readable"""
+    return string.replace(",<", "<")
+
+
+def remove_whitespace_from_equals(string):
+    """receives string and returns it without inline comments"""
+    return string.replace(" =", "=").replace("= ", "=")
+
+
+def remove_comments(string):
+    """receives string and returns it without inline comments"""
+    return re.sub(re.compile("#.*?\n"), "", string)
```

Comment:
this reminds me that you might want to check for commented requirements. That is, some people might have them and  you should probably make sure they're removed. Not a big deal.. but it would be nice to handle all cases.